### PR TITLE
Add support for cleaning StatefulSets with PVCs

### PR DESCRIPTION
**Pull request by raminqaf**

### CODE REVIEW

**Review by philipp94831 — DISMISSED** [review]

File: kpops/components/streams_bootstrap/streams/streams_app.py [hunk 1]
```diff
@@ -24,6 +30,26 @@ class StreamsAppCleaner(KafkaAppCleaner):
...
+        await super().clean(dry_run)
+        if self.app.stateful_set and self.app.persistence.enabled:
+            await self.clean_pvcs(dry_run)
+
+    async def clean_pvcs(self, dry_run: bool) -> None:
+        if dry_run:
+            pvc_names = await self.pvc_handler.get_pvc_names()
+            log.info(
+                f"Deleting the PVCs {pvc_names} for StatefulSet '{self.full_name}'"
+            )
```
Comment (disrupted) [inline kpops/components/streams_bootstrap/streams/streams_app.py:48]: how about moving the dry_run case to the PVC handler as well?

Comment (raminqaf) [reply kpops/components/streams_bootstrap/streams/streams_app.py:48]: I would disagree. I would keep the functionality of the PVCHandler limited to Kubernetes and night tight to KPOps dry_run.

File: kpops/component_handlers/kubernetes/pvc_handler.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        self.app_name = app_name
+        self.namespace = namespace
+
+    @classmethod
+    async def create(cls, app_name: str, namespace: str) -> PVCHandler:
+        self = cls(app_name, namespace)
+        await config.load_kube_config()
+        return self
+
+    async def get_pvc_names(self) -> list[str]:
```
Comment (disrupted) [inline kpops/component_handlers/kubernetes/pvc_handler.py:22]:
```suggestion
    async def list_pvcs(self) -> list[str]:
```

Comment (disrupted) [reply kpops/component_handlers/kubernetes/pvc_handler.py:22]: more in line with your other method

**Review by disrupted — COMMENTED** [review]
nitpick, otherwise LGTM

File: kpops/component_handlers/kubernetes/pvc_handler.py (hunk 2 above)
Comment (raminqaf) [reply kpops/component_handlers/kubernetes/pvc_handler.py]: done!

**Pull request merged**